index.html

```diff
@@ -1028,6 +1061,461 @@ <h2>
+        </dt>
+        <dd>
+          <p>
+            Array of {{Gamepad/GamepadHapticEffectType}} values representing all
+            the types of haptic effects that the actuator supports. This
+            property MUST contain all the {{Gamepad/GamepadHapticEffectType}}
+            values that the actuator supports, unless the [=user agent=] does
+            not support playing effects of that type. The same object MUST be
+            returned and MUST NOT change for the lifetime of the
+            {{GamepadHapticActuator}}.
+          </p>
+          <p>
+            The {{GamepadHapticActuator/effects}} getter steps are:
+          </p>
+          <ol>
+            <li>Return [=this=].{{GamepadHapticActuator/[[effects]]}}.
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          <dfn>playEffect()</dfn> method
+        </dt>
+        <dd>
+          <p>
+            The {{GamepadHapticActuator/playEffect()}} method steps are:
+          </p>
+          <ol>
+            <li>If |params:GamepadEffectParameters| does not describe a [=valid
+            effect=] of type |type:GamepadHapticEffectType|, return [=a promise
+            rejected with=] reason {{TypeError}}.
+            </li>
+            <li>Let |document| be the [=current settings object=]'s [=relevant
+            global object=]'s [=associated `Document`=].
+            </li>
+            <li>If |document| is `null` or |document| is not [=Document/fully
+            active=] or |document|'s [=Document/visibility state=] is `"hidden"`, return
+            [=a promise resolved with=] {{GamepadHapticsResult/"preempted"}}.
+            </li>
+            <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+            is not `undefined`:
+              <ol>
+                <li>Let |effectPromise| be
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}.
+                </li>
+                <li>Set
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
+                `undefined`.
+                </li>
+                <li>[=Queue a global task=] on the [=relevant global object=]
+                of [=this=] using the [=gamepad task source=] to [=resolve=]
+                |effectPromise| with {{GamepadHapticsResult/"preempted"}}.
+                </li>
+              </ol>
+            </li>
+            <li>If this actuator cannot [=play effects with type=] |type|,
+            return [=a promise rejected with=] reason {{NotSupportedError}}.
+            </li>
+            <li>Let {{GamepadHapticActuator/[[playingEffectPromise]]}} be [=a
+            new promise=].
+            </li>
+            <li>Do the following steps in parallel:
+              <ol>
+                <li>[=Issue a haptic effect=] to the actuator with |type| and
+                |params|.
+                </li>
+                <li>When the effect completes, if it was not preempted, [=queue
```

Comment:
Let's change "if it was not preempted" to an explicit check for `GamepadHapticActuator/[[playingEffectPromise]]`

Reply:
Thinking a little bit more about it, I think that only checking `GamepadHapticActuator/[[playingEffectPromise]]` might not be the best way to proceed. For example:
1. Effect `e1` starts playing. So `GamepadHapticActuator/[[playingEffectPromise]]` is set to `p1`
2. While `e1` is already executing (gamepad is already vibrating), we issue `e2`
3. `GamepadHapticActuator/[[playingEffectPromise]]` is set to `p2` and `p1` is resolved with `"preempted"`.
4. `e1` finishes playing normally and `GamepadHapticActuator/[[playingEffectPromise]]` is already set to `p2`. `p2` is resolved with `"completed"` and `GamepadHapticActuator/[[playingEffectPromise]]` is set to `null`. 

I think that we could maybe add an explicit "stop haptics effect", like we did with `reset()`:
```
<li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
            is not `null`:
              <ol>
                <li>Let |effectPromise| be
                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}.
                </li>
                <li>Set
                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
                `null`.
                </li>
                <li>[=Stop haptic effects=] on [=this=]'s gamepad's actuator.
               </li>
                <li>[=Queue a global task=] on the [=relevant global object=]
                of [=this=] using the [=gamepad task source=] to [=resolve=]
                |effectPromise| with {{GamepadHapticsResult/"preempted"}}.
                </li>
              </ol>
            </li>
```
I think that is why the "if it was not preempted" part is also relevant. What do you think? Also @nondebug.